scripts/python/phdf_diff.py

```diff
@@ -64,8 +66,120 @@ def addPath():
     #sys.path.insert(0,myPath+'/../vis/python')
     #sys.path.insert(0,myPath+'/vis/python')
 
-def compare(files, all=False, brief=True, quiet=False, one=False, tol=1.0e-12):
-    """ compares two hdf files """
+def compare_metadata( f0, f1, tol=1.0e-12):
+    """ compares metadata of two hdf files f0 and f1. Returns 0 if the files are equivalent.
+
+        Error codes:
+            10 : Times in hdf files differ
+            11 : Attribute names in Info of hdf files differ
+            12 : Values of attribute names in Info of hdf files differ
+            13 : Meta variables (Locations, VolumeLocations, LogicalLocations, Levels) differ
+    """
+    ERROR_TIME_DIFF=10
+    ERROR_INFO_ATTRS_DIFF=12
+    ERROR_INFO_VALUES_DIFF=13
+    ERROR_META_VARS_DIFF=14
+
+    #Check in the metadata in Info is identical, except use tol for comparing "Time"
+    f0_Info = { key:value for key,value in f0.Info.items() if key != "Time" and key != "BlocksPerPE" }
+    f1_Info = { key:value for key,value in f1.Info.items() if key != "Time" and key != "BlocksPerPE" }
+
+    errTime = np.abs(f0.Time- f1.Time)
+    if errTime > tol:
+        print(f"""
+        Time of outputs differ by {f0.Time - f1.Time}
+
+        Quitting...
+        """)
+        return(ERROR_TIME_DIFF)
+    if f0_Info.keys() != f1_Info.keys():
+        print("""
+        Attributes in '/Info' of outputs differ
+
+        Quitting...
+        """)
+        return(ERROR_INFO_ATTRS_DIFF)
+
+    info_diffs = list(np.any(f0_Info[key] != f1_Info[key]) for key in f0_Info.keys())
+    infos_differ = np.any( info_diffs )
+    if infos_differ:
+        print("\n'/Info' of outputs differ\n")
+        print("Differing attributes: ",list( (key) for key in f0_Info.keys() if np.any(f0_Info[key] != f1_Info[key]))  )
+        print("\nQuitting...\n")
+        return(ERROR_INFO_VALUES_DIFF)
+    else:
+        print('  %20s: no diffs'%"Info")
+
+    # Now go through all variables in first file
+    # and hunt for them in second file.
+    #
+    # Note that indices don't match when blocks
+    # are different
+    no_meta_variables_diff = True
+
+    otherBlockIdx = list(f0.findBlockIdxInOther(f1,i) for i in range(f0.NumBlocks))
+
+    for var in set(f0.Variables+f1.Variables):
+        if var in ['Locations', 'VolumeLocations']:
+            for key in f0.fid[var].keys():
+                #Compare raw data of these variables
+                val0 = f0.fid[var][key]
+                val1 = f1.fid[var][key]
+
+                #Sort val1 by otherBlockIdx
+                val1 = val1[otherBlockIdx]
```

Comment:
`otherBlockIdx` gives the indexing of mesh blocks on `f1` in the same order as the meshblocks in `f0`, since meshblocks are free to rearrange in order due to changing MPI ranks.